notebooks/Distance.ipynb

```diff
@@ -0,0 +1,812 @@
+   "source": [
+    "# The distance from source to target node\n",
+    "print(dijkstra.distance(target))\n",
+    "# The number of shortest paths between the source node\n",
+    "print(dijkstra.numberOfPaths(target))\n",
+    "# Returns a shortest path from source to target\n",
+    "print(dijkstra.getPath(target))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Bidirectional Dijkstra"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The Bidirectional Dijkstra algorithm explores the graph from both the source and target nodes until the two explorations meet.  This version of Dijkstra is more efficient than the convential Dijkstra when the target node is known.\n",
+    "\n",
+    "The [BidirectionalDijkstra(G, source, target, storePred=True)](https://networkit.github.io/dev-docs/python_api/distance.html?highlight=bidirec#networkit.distance.BidirectionalDijkstra) constructor expects a graph, the source and target nodes as mandatory parameters. The algorithm will also store the predecessors and reconstruct a shortest path from the source and the target if `storePred` is true."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize algorithm\n",
+    "biDij = nk.distance.BidirectionalDijkstra(G, source, target)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run\n",
+    "biDij.run()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The distance from source to target node\n",
+    "print(biDij.getDistance())\n",
+    "# The path from source to target node\n",
+    "print(biDij.getPath())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Commute Time Distance"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This class computes the Euclidean Commute Time Distance between each pair of nodes for an undirected unweighted graph.\n",
+    "\n",
+    "The [CommuteTimeDistance(G, tol=0.1)](https://networkit.github.io/dev-docs/python_api/distance.html?highlight=commute#networkit.distance.CommuteTimeDistance) constructor expects a graph as a mandatory parameter. The optional parameter `tol` is the tolerance parameter used for approximation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize algorithm\n",
+    "GUndirected = G.toUndirected()\n",
+    "ctd = nk.distance.CommuteTimeDistance(GUndirected)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run\n",
+    "ctd.run()\n",
+    "# Alternatively \n",
+    "# ctd.runSinglePair(source,target)"
```

Comment:
You can run this in another cell, and mention that if one wants to compute the commute time distance between two nodes, then they should use `runSinglePair`.